pkg/integrations/squid_exporter/squid_exporter.go

```diff
@@ -0,0 +1,111 @@
+package squid_exporter
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"strconv"
+
+	se "github.com/boynux/squid-exporter/collector"
+	"github.com/go-kit/log"
+	"github.com/grafana/agent/pkg/integrations"
+	"github.com/prometheus/client_golang/prometheus"
+
+	integrations_v2 "github.com/grafana/agent/pkg/integrations/v2"
+	"github.com/grafana/agent/pkg/integrations/v2/metricsutils"
+)
+
+// DefaultConfig is the default config for the squid integration
+var DefaultConfig = Config{
+	Address:  "localhost:3128",
+	Username: "",
+	Password: "",
+}
+
+var (
+	errNoAddress  = errors.New("no address was provided")
+	errNoHostname = errors.New("no hostname in provided address")
+	errNoPort     = errors.New("no port in provided address")
+)
+
+// Config is the configuration for the squid integration
+type Config struct {
+	Address  string `yaml:"address"`
+	Username string `yaml:"username"`
+	Password string `yaml:"password"`
+	Host     string
+	Port     int
+}
+
+func (c *Config) validate() error {
+	if c.Address == "" {
+		return errNoAddress
+	}
+
+	host, scheme, err := net.SplitHostPort(c.Address)
+	if err != nil {
+		return err
+	}
+
+	if host == "" {
+		return errNoHostname
+	}
+	c.Host = host
+
+	if scheme == "" {
+		return errNoPort
+	}
+
+	port, _ := strconv.Atoi(scheme)
```

Comment:
Hey Keith, do you mean like ignoring the error portion?

Reply:
Yeah the `Atoi` call returns an error which we are specifically ignoring on this line in the validate. Just feels like if it fails we should return an error rather than ignoring it and getting a potential panic somewhere down the road.